File: server/src/main/scala/kpn/api/common/route/RouteInfo.scala
package kpn.api.common.route

import kpn.api.common.Bounds
import kpn.api.common.RouteType

case class RouteInfo(
  routeId: Long,
  routeName: String,
  routeTypes: Seq[RouteType],
  memberCount: Long,
  pathCount: Long,
  segmentCount: Long,
  changeCount: Long,
  bounds: Option[Bounds],
)

File: backend/api/src/main/java/kpn/api/common/route/RouteInfo.java
package kpn.api.common.route;

import kpn.api.common.Bounds;
import kpn.api.common.RouteType;

import java.util.Optional;
import com.google.common.collect.ImmutableList;

public record RouteInfo(
  Long routeId,
  String routeName,
  ImmutableList<RouteType> routeTypes,
  Optional<Bounds> bounds,
  Long memberCount,
  Long pathCount,
  Long segmentCount,
  Long changeCount
) {}
